### PR TITLE
Add plain properties conformance test

### PR DESCRIPTION
**Pull request by Frassle**
Adds a new provider which takes all its inputs as plain values to test that languages can specify plain inputs.

### CODE REVIEW

File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/tsc/sdks/plain-13.0.0/types/input.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// *** WARNING: this file was generated by pulumi-language-nodejs. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+import * as pulumi from "@pulumi/pulumi";
+import * as inputs from "../types/input";
+import * as outputs from "../types/output";
+
+export interface DataArgs {
+    boolArray: boolean[];
```
Comment (justinvp) [inline sdk/nodejs/cmd/pulumi-language-nodejs/testdata/tsc/sdks/plain-13.0.0/types/input.ts:9]: Do we have any conformance tests where the container is plain but the values are inputs?
- `Input<boolean>[]`
- `{[key]: string]: Input<string>}`

It's common to want to do that for packaged components (MLCs) so you can loop over the container to create child resources without needing to do it inside an `apply`.

File: cmd/pulumi-test-language/providers/plain_provider.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+			"plain:index:Data": {
+				ObjectTypeSpec: schema.ObjectTypeSpec{
+					Type:       "object",
+					Properties: dataProperties,
+					Required:   dataRequired,
+				},
+			},
+		},
+		Resources: map[string]schema.ResourceSpec{
+			"plain:index:Resource": {
```
Comment (justinvp) [inline cmd/pulumi-test-language/providers/plain_provider.go:141]: It's a little unusual to have plain values for a custom resource (they're primarily for components), but I guess someone may want to do it and it should work, so might as well test it. 👍

**Review by justinvp — APPROVED** [review]

File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/tsc/sdks/plain-13.0.0/types/input.ts (hunk 1 above)
Comment (Frassle) [reply sdk/nodejs/cmd/pulumi-language-nodejs/testdata/tsc/sdks/plain-13.0.0/types/input.ts:9]: No but we could add that, would probably make sense to just extend this test?

File: cmd/pulumi-test-language/providers/plain_provider.go (hunk 2 above)
Comment (Frassle) [reply cmd/pulumi-test-language/providers/plain_provider.go:141]: Good point, we should check this works as a component as well.

**Pull request merged**